### PR TITLE
247 mso schema site anp epg domain 4 2 attributes

### LINKED ISSUE #247: Update mso_schema_site_anp_epg_domain to support all vmm domain attributes in ndo 4.2+
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add new supported parameters for vmm domain association:

delimeter
custom_epg_name
mac_changes
forget_transmits
netflow_pref
num_port
port_allocation
binding_type (required)
allow_promiscuous

### New or Affected Resource(s) + MSO Class(es):

* mso_schema_site_anp_epg_domain

### MSO version

* V 4.2.x

### APIC version and APIC Platform for Site level Resources

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #247

### CODE REVIEW

**Comment by akinross** [comment]
mso-go-client update is still pending, waiting on PR to get merged

File: website/docs/r/schema_site_anp_epg_domain.html.markdown [hunk 1]
```diff
@@ -35,35 +35,68 @@ resource "mso_schema_site_anp_epg_domain" "site_anp_epg_domain" {
...
-  resolution_immediacy = "lazy"
-  vlan_encap_mode = "static"
-  allow_micro_segmentation = true
-  switching_mode = "native"
-  switch_type = "default"
-  micro_seg_vlan_type = "vlan"
-  micro_seg_vlan = 46
-  port_encap_vlan_type = "vlan"
-  port_encap_vlan = 45
+
```
Comment (shrsr) [inline website/docs/r/schema_site_anp_epg_domain.html.markdown:45]: It might be good to add: ### domain_dn usage in version < 4.2 ###

**Review by shrsr — CHANGES_REQUESTED** [review]

File: website/docs/r/schema_site_anp_epg_domain.html.markdown (hunk 1 above)
Comment (akinross) [reply website/docs/r/schema_site_anp_epg_domain.html.markdown:45]: added also in examples

**Review by sajagana — CHANGES_REQUESTED** [review]

File: examples/schema_site_anp_epg_domain/main.tf [hunk 2]
```diff
@@ -13,44 +13,171 @@ provider "mso" {
...
+  port_encap_vlan          = 45
+  delimiter                = "|"
+  binding_type             = "static"
+  port_allocation          = "fixed"
+  num_ports                = 3
+  netflow                  = "disabled"
+  allow_promiscuous        = "accept"
+  mac_changes              = "reject"
+  forged_transmits         = "reject"
+  custom_epg_name          = "custom_epg_name_1"
```
[the remark this answers is not in the document]
Comment (akinross) [reply examples/schema_site_anp_epg_domain/main.tf:110]: please check again was specific config combi in our setup

**Review by shrsr — APPROVED** [review]
LGTM

**Review by sajagana — DISMISSED** [review]
LGTM!

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**